### PR TITLE
*: use fileutil.TouchDirAll

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: etcdctl/ctlv2/command/backup_command.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func handleBackup(c *cli.Context) {
 		destWAL = path.Join(c.String("backup-dir"), "member", "wal")
 	}
 
-	if err := os.MkdirAll(destSnap, 0700); err != nil {
+	if err := fileutil.TouchDirAll(destSnap); err != nil {
```
Comment (xiang90) [inline etcdctl/ctlv2/command/backup_command.go]: we should still use mkdirall when we want to ensure we create a dir. touchdir will not return an error if the dir already exists.

File: etcdserver/server.go [hunk 2]
```diff
@@ -246,7 +243,7 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 
 	haveWAL := wal.Exist(cfg.WALDir())
 
-	if err = os.MkdirAll(cfg.SnapDir(), privateDirMode); err != nil && !os.IsExist(err) {
+	if err = fileutil.TouchDirAll(cfg.SnapDir()); err != nil && !os.IsExist(err) {
```
Comment (xiang90) [inline etcdserver/server.go]: remove the os.IsExist checking. touchdir already checks that.

File: etcdserver/storage.go [hunk 3]
```diff
@@ -129,7 +130,7 @@ func makeMemberDir(dir string) error {
 	case !os.IsNotExist(err):
 		return err
 	}
-	if err := os.MkdirAll(membdir, 0700); err != nil {
+	if err := fileutil.TouchDirAll(membdir); err != nil {
```
Comment (xiang90) [inline etcdserver/storage.go]: use mkdir. we want to ensure we make a dir.

**Comment by xiang90** [comment]
there are a bunch of places that we should use makedir instead of touchdir since we want to ensure we successfully make a dir. please change back.

**Comment by gyuho** [comment]
@xiang90 Just fixed up only to use in server.go.

> touchdir will not return an error if the dir already exists.

I think if we want that behavior, we should use `os.Mkdir` instead?
We already use `os.MkdirAll` that doesn't return errors even when the directory already exists.

```go
err := os.MkdirAll("test", 0700)
// nil

err = os.MkdirAll("test", 0700)
// nil

err = os.Mkdir("test", 0700)
// mkdir test: file exists
```

Thanks!

**Comment by xiang90** [comment]
lgtm. and you are right. probably we should make some of the code more strict.

**Comment by gyuho** [comment]
I will double-check other parts, and make them consistent next time. Merging this first because it's not urgent. Thanks!

**Pull request merged**